File: backend/src/main/java/ResumeBuilder/backend/Models/DatabaseModels/EducationModel.java
package ResumeBuilder.backend.Models.DatabaseModels;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import ResumeBuilder.backend.Interfaces.IWriter;
import ResumeBuilder.backend.Utilities.DateFormatUtility;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "education")
public class EducationModel implements IWriter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public int id;

    @Column
    public String school;

    @Column
    public String degree;

    @Column
    public String location;

    @Column
    public LocalDate startDate;

    @Column
    public LocalDate endDate;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "education_id")
    public List<EducationBulletPointModel> description;

    public Object get(String key) {
        if (key.equals("school")) {
            return school;
        } else if (key.equals("startDate")) {
            return DateFormatUtility.toString(startDate);
        } else if (key.equals("endDate")) {
            return DateFormatUtility.toString(endDate);
        } else if (key.equals("degree")) {
            return degree;
        } else if (key.equals("location")) {
            return location;
        } else if (key.equals("description")) {
            return description;
        }
        return null;
    }
}

File: frontend/models/EducationModel.js
import EducationBulletPointModel from "./EducationBulletPointModel";
import { v4 as uuid } from "uuid";

export default class EducationModel {
  id = uuid();
  school = "";
  degree = "";
  location = "";
  startDate;
  endDate;
  description = new Array();

  constructor(data) {
    if (data != null) {
      this.id = data.id;
      this.school = data.school;
      this.degree = data.degree;
      this.location = data.location;
      this.startDate = data.startDate;
      this.endDate = data.endDate;
      this.description = data.description.map((item) => {
        return new EducationBulletPointModel(item);
      });
    }
  }
}
